extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/PentahoMetadataDomainRepository.java

```diff
@@ -675,6 +823,82 @@ private void internalReloadDomains() {
     }
   }
 
+  /**
+   * POC - DEBUG
+   * Only migrate if certain properties don't exist and other conditions.
+   * Else return file metadata.
+   * @param serializableId
+   * @return
+   */
+  public Map<String, Serializable> getFileMetadataHelper( Serializable serializableId ) {
+    Map<String, Serializable> fileMetadata = repository.getFileMetadata( serializableId );
+
+    // check for domain, has not been previously migrated, flag set to migrate
+    if ( isDomain( fileMetadata ) && !hasDatasourceType( fileMetadata ) && canMigrate() ) {
```

Comment:
[#6 professionalize code for migrating domain](https://jira.pentaho.com/browse/BACKLOG-36130)